From 93383be4f6dfb29644fd7d46ba81243f4d7f5c06 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 30 Aug 2018 15:58:15 +0800
Subject: [PATCH 4524/5242] MLK-19413-13 gpu: imx: dpu: common: Specify
 has_syncmode_fixup flag for DPU variants

commit  db33997cfaf4a58591e25cde610cd8a11260972c from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch introduces a new has_syncmode_fixup entry in struct dpu_devtype and
specifies this flag for DPU variants found in existing SoCs.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    3 +++
 drivers/gpu/imx/dpu/dpu-prv.h    |    1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index ef7380a..a2d35f8 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -544,6 +544,7 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 	.has_prefetch = false,
 	.has_disp_sel_clk = false,
 	.has_dual_ldb = false,
+	.has_syncmode_fixup = false,
 	.pixel_link_quirks = false,
 	.pixel_link_nhvsync = false,
 	.version = DPU_V1,
@@ -574,6 +575,7 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 	.has_prefetch = true,
 	.has_disp_sel_clk = true,
 	.has_dual_ldb = false,
+	.has_syncmode_fixup = true,
 	.pixel_link_quirks = true,
 	.pixel_link_nhvsync = true,
 	.version = DPU_V2,
@@ -604,6 +606,7 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 	.has_prefetch = true,
 	.has_disp_sel_clk = false,
 	.has_dual_ldb = true,
+	.has_syncmode_fixup = false,
 	.pixel_link_quirks = true,
 	.pixel_link_nhvsync = true,
 	.version = DPU_V2,
diff --git a/drivers/gpu/imx/dpu/dpu-prv.h b/drivers/gpu/imx/dpu/dpu-prv.h
index 3bfcaad..e53b529 100644
--- a/drivers/gpu/imx/dpu/dpu-prv.h
+++ b/drivers/gpu/imx/dpu/dpu-prv.h
@@ -209,6 +209,7 @@ struct dpu_devtype {
 	bool has_prefetch;
 	bool has_disp_sel_clk;
 	bool has_dual_ldb;
+	bool has_syncmode_fixup;
 	bool pixel_link_quirks;
 	bool pixel_link_nhvsync;	/* HSYNC and VSYNC high active */
 	unsigned int version;
-- 
1.7.9.5

